### PR TITLE
Added enable_ibrowse_traces flag for ibrowse debugs

### PR DESCRIPTION
**Pull request by jashaik**
Signed-off-by: jan shahid shaik <jashaik@progress.com>

### Description

Added enable_ibrowse_traces for enabling ibrowse debugs as part of Improve logging and tooling for troubleshooting.

### Issues Resolved

https://app.zenhub.com/workspaces/chef-infra-server-team-5fc64867d45ca500173dbbc7/issues/chef/chef-server/2858

### Check List

- [ ] New functionality includes tests
- [ ] All buildkite tests pass
- [ ] Full omnibus build and tests in buildkite pass
- [ ] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/main/CONTRIBUTING.md#developer-certification-of-origin-dco>
- [ ] PR title is a worthy inclusion in the CHANGELOG

### CODE REVIEW

**Comment by jashaik** [comment]
Pipeline Information -
Adhoc - https://buildkite.com/chef/chef-chef-server-main-omnibus-adhoc/builds/3416

**Comment by tas50** [comment]
Let's make sure to also update the docs to document this new config.

File: src/oc_erchef/rebar.lock [hunk 1]
```diff
@@ -68,7 +68,7 @@
  {<<"goldrush">>,{pkg,<<"goldrush">>,<<"0.1.9">>},1},
  {<<"ibrowse">>,
   {git,"git://github.com/chef/ibrowse.git",
-       {ref,"1df08a6aca30614f69126451957af6a3d0aeb7b2"}},
+       {ref,"c4e319e193410e8e7527c5f43563c46a0718bca8"}},
```
Comment (PrajaktaPurohit) [inline src/oc_erchef/rebar.lock:71]: Lets keep on the same version for bookshelf and bifrost.

Comment (jashaik) [reply src/oc_erchef/rebar.lock:71]: bifrost is not using it and Bookshelf doesn’t have the direct usage of ibrowse except some test bookshelf_bench.erl which is pulling the ibrowse from `git://github.com/cmullaparthi/ibrowse.git`.
Please let me know If I still need to pass this flag from ominibus to bookshelf.
Screenshot 2021-11-16 at 11 20 15 AM

**Comment by jashaik** [comment]
ibrowse PR - https://github.com/chef/ibrowse/pull/5
@PrajaktaPurohit

**Comment by lbakerchef** [comment]
@jashaik Could we see a screenshot or a dump of this working?  Enable it and show it working, then disable it and show it not working?  Thanks.

**Comment by jashaik** [comment]
> @jashaik Could we see a screenshot or a dump of this working? Enable it and show it working, then disable it and show it not working? Thanks.

This was tested already on the Facebook server and got the traces also. I cherry-picked those commit.

File: src/oc_erchef/rebar.lock (hunk 1 above)
Comment (vinay-satish) [reply src/oc_erchef/rebar.lock:71]: Now that the changes for ibrowse are merged to main, should we use that merge commit hash?

Comment (PrajaktaPurohit) [reply src/oc_erchef/rebar.lock:71]: yeah this needs to be the hash in the chef/ibrowse repo with the corresponding changes for now.

**Comment by lbakerchef** [comment]
After rebase, and updating SHAs for ibrowse:
https://buildkite.com/chef/chef-chef-server-main-omnibus-adhoc/builds/3499
https://buildkite.com/chef/chef-umbrella-main-chef-server/builds/922

**Pull request merged**